ex3.py

```diff
@@ -0,0 +1,45 @@
+import pickle, base64
+import os
+import datetime
+
+def rename_files(folder_name, key, d):
+    for file_name in os.listdir(path + folder_name + '/'):
+        num = file_name.split('_')
+        for i in d[key]:
+            if(str(num[1])[:-4] == str(i)):
+                my_source = path + folder_name + '/' + str(num[0]) + '_' + str(num[1])
+                unixToDatetime = datetime.datetime.fromtimestamp(int(num[0]) + int(d[key][i]))
+                my_dest = path + folder_name + '/' + str(unixToDatetime.date()) + ".npz"
+                os.rename(my_source, my_dest)
+
+
+dict_list = []
+dictionary = {} 
+# open the sixth file and append all the dicts to a dict_list.
+with (open("ex3/6/6.pkl", "rb")) as openfile:
+    while True:
+        try:
+            dict_list.append(pickle.load(openfile))
+        except EOFError:
+            break
+
+# converts keys in dict_list.
+for d in dict_list:
+    for key in d:
+        new_key = (str(base64.b32decode(key))[2:-1])
+        dictionary[new_key] = d[key]
```

Comment:
Bad practice, you should create this dict for every  iteration. Also create it closer to where its used, especially when it acts more like a temp variable.  

What if not all the keys are the same?